### DISCUSSION TITLE
Pre-charging microinverter capacitors?

### DISCUSSION THREAD

**Discussion by greymda** [hoylabs/OpenDTU-OnBattery#886] [Q&A]
Hey community, hope your wisdom can help. I have a HM-800 inverter and building a DIY 1kwh battery for start (16 lifepo4 20Ah cells in 1 series) and i am finding contradictory information on the internet. Somewhere it's stated that pre-charging capacitors are needed only for 2000W+ like here

> You need to precharge any inverter 2000W or above. It’s important to do this any time that you’ve disconnected the batteries for a prolonged period. Smaller inverters have smaller capacitors that charge even quicker than those in larger inverters. This means that the current flow slows quickly enough to avoid any BMS damage and large sparks. Therefore, it’s safe to connect your inverter directly to your busbars.

somewhere that it's a must have!

What is this community experience and consensus?

**Comment by schlimmchen** [comment]
Always pre-charge using a resistor and be safe :+1:

Have a loot at #528

**Reply by greymda** [reply]
@schlimmchen thank you, time for google translate! lately it's a "always-on tab"! 
one more question: if to use a resistor, is it a one-time usage when physically connecting/disconnecting the battery cables OR say when DPL works and it shuts down the inverter i need somehow this resistor?

**Reply by spcqike** [reply]
It’s a one time use after physical disconnect. 

As long as the inverter is connected to the battery, the capacitors stay charged. So a stopped inverter (manually or by DPL) is still connected and charged (but also drains the battery, as the inverter powers itself from DC) 

so whenever you disconnect the battery (plug it out or BMS shutdown), you need to manually charge the inverter.

**Reply by schlimmchen** [reply]
> is it a one-time usage when physically connecting/disconnecting the battery cables

Yes. This includes intervention of the BMS or fuses, etc.

> say when DPL works and it shuts down the inverter i need somehow this resistor?

No. The DPL only sends the shutdown command, the capacitors stay physically connected to the battery and stay charged.

Note that you **must** remove the resistor after pre-charging the capacitors.

After a disconnect, I use a fuse/switch to connect the inverter to the battery with a resistor (~1Ohm) in series. Pre-charging does not take long (give it two seconds). I then enable another fuse/switch, which establishes the direct connection. The pre-charge-switch **must** be disconnected afterwards.

**Reply by greymda** [reply]
just for my understanding @spcqike & @schlimmchen - when you say "BMS intervention" does this include cases like when BMS finds overcharging or you mean like when i disable the BMS by its power switch? thanks

**Reply by schlimmchen** [reply]
If the BMS cuts the connection to protect the battery, e.g., because of overcharging, you will need to pre-charge, unless your charge controller is still alive and outputs a respective voltage/current to feed the inverter.

This is a tricky problem. The consensus is to avoid BMS intervention (for a couple of reasons) whenever possible.

I had the BMS reconnect the battery after some problem went away by itself, and the fuse blew. Not ideal.

**Reply by swingstate** [reply]
I am using a HM1500 and during the time owning it I did approx 10 - 20 DC disconnects and reconnects without any capacitors or other precharge solutions. I always disconnect the AC side of the Inverter first and then the DC with a DC Switch between Batteries and the Busbars. To reconnect, I connect the DC side first prior the AC side.  So far no issues. I always try to keep the DC disconnects very short, not sure for how long the capacitors inside the inverter hold any charge.

**Reply by greymda** [reply]
> The consensus is to avoid BMS intervention (for a couple of reasons) whenever possible.

any tips to accomplish this?

**Reply by schlimmchen** [reply]
You will find them by yourself. Have this example: Chose an overvoltage protection value that is safe for your cells, but set your charge controller to charge only to less than that. You overvoltage protection should never trigger unless something goes south.

**Reply by swingstate** [reply]
Act before the BMS has to. Basically switch off the Inverter before you hit the zone where the BMS would kick in. I use Homeassistant for some automations in that area and also the DTU to enable/disable the inverter.

**Reply by greymda** [reply]
gotcha, but how about if during night the battery gets drained out? as we're speaking about something around 1kwh max. will this be a problem?

**Reply by swingstate** [reply]
switch the inverter off at for example 15 or 20% SoC or the Voltage equivalent - depending on your battery. 16 cells sounds like a 48V Battery. I used these charts for the PowerQueen I was using prior. https://cleversolarpower.com/lifepo4-voltage-chart/
At the end, you should measure the voltage of your cells and switch the inverter off at e.g. 3.1V / cell respectively at 48V. But that's just one example. Other people might tell you different things.

**Reply by greymda** [reply]
I think i understand. it will be 16 cells[ EVE C40 40135 - 20Ah](https://eu.nkon.nl/rechargeable/lifepo4/overige-formaten/eve-c40-40135-20ah-3-2v.html) - and a JK BMS.
